ui/src/ElasticSearch.elm

```diff
@@ -0,0 +1,134 @@
+module ElasticSearch exposing (..)
+
+import Http
+import Decoder exposing (buildRecordDecoder)
+import Json.Decode as Decode
+import Json.Encode as Encode
+import Types exposing (..)
+
+
+filterToJsonProperty : String -> Maybe String
+filterToJsonProperty filter =
+    if filter == "product" then
+        Just "source.product"
+    else if filter == "channel" then
+        Just "target.channel"
+    else if filter == "locale" then
+        Just "target.locale"
+    else if filter == "version" then
+        Just "target.version"
+    else if filter == "platform" then
+        Just "target.platform"
+    else
+        Nothing
+
+
+encodeMustClause : Filters -> Encode.Value
+encodeMustClause { product, channel, locale, version, platform, buildId } =
+    let
+        encodeFilter ( name, value ) =
+            Encode.object
+                [ ( if name == "build.id" then
+                        "prefix"
+                    else
+                        "match"
+                  , Encode.object [ ( name, Encode.string value ) ]
+                  )
+                ]
+
+        refineFilters ( k, v ) acc =
+            if k == "buildId" && v /= "" then
+                ( "build.id", v ) :: acc
+            else
+                case filterToJsonProperty k of
+                    Just property ->
+                        if v /= "all" then
+                            ( property, v ) :: acc
+                        else
+                            acc
+
+                    Nothing ->
+                        acc
+    in
+        [ ( "buildId", buildId )
+        , ( "product", product )
+        , ( "channel", channel )
+        , ( "locale", locale )
+        , ( "version", version )
+        , ( "platform", platform )
+        ]
+            |> List.foldr refineFilters []
+            |> List.map encodeFilter
+            |> Encode.list
+
+
+encodeQuery : Filters -> Int -> Int -> Encode.Value
+encodeQuery filters size page =
```

Comment:
How about renaming the variable `size` to `pageSize` or `resultsPerPage` or `limit` or something?